### ISSUE TITLE
Authorisation Error Since January 24 Update (Error 403: rate_limit_exceeded)

### ISSUE THREAD

**Issue by cozumel68** [kewisch/gdata-provider#381]
**Edit:** Forgot to say that I'm using:
Thunderbird Version 91.5.0
Provider for Google Calendar 91.0.3
Windows 10 Pro 21H2 19044.1466 with Feature Experience Pack 120.2212.3920.0
**EndEdit**

Since the 24th January update, I have not been able logon to the 'Provider for Google Calendar' Thunderbird add-on.  This may possibly be due to the Google Cloud Platform Console identifying the add-on app as still being in the testing phase and therefore limited to 100 max users.  If that is the case, then the software would only become available to all users, once it has been identified as published.....If it helps, I get the following text in the error dialog box:

Authorisation Error
Error 403: rate_limit_exceeded

This app has reached its sign-in rate limit for now.

Google limits how quickly an app can get new users. You can try signing in again later or ask the developer (provider-for-google-calendar@googlegroups.com) to increase this app's sign-in rate limit.

If you are the developer for this app, you can request a sign-in rate limit increase.
Learn more
The content in this section has been provided by the app developer. This content has not been reviewed or verified by Google.
If you’re the app developer, make sure that these request details comply with Google policies.

    login_hint: 0809cc52-1df5-438a-b810-1c2d0d9cd95d
    hl: en-GB
    response_type: code
    redirect_uri: urn:ietf:wg:oauth:2.0:oob:auto
    client_id: 647770275422-f30mivm6f2nli7qu4mpt20a3lc83t1vv.apps.googleusercontent.com
    access_type: offline
    scope: https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/tasks

**Comment by mario-grtr** [comment]
Same Issue with Thunderbird v 91.5.1

**Comment by picorealm** [comment]
I also have this issue.  I unsubscribed the calendar and tried to re-subscribe but the error prevents me from being able to authorize and add the calendar.
Thunderbird  91.5.0 20220106182030
Linux 5.15.0-0.bpo.2-amd64 #1 SMP Debian 5.15.5-2~bpo11+1 (2022-01-02)
Provider for Google Calendar 91.0.4

**Comment by cozumel68** [comment]
Okay then, so I've just read in post #373, in a reply from kewisch (the owner of this repository), that the problem has been identified.  That an incorrect API was used for the 91.0.3 release, resulting in users having to login again and user rate limits therefore becoming exceeded.   kewisch stated that this problem will be rectified for all users shortly and to hang in and be patient...

Another reply from pajew in the same post #373, simply stated that this add-on is no longer necessary.  So I checked this out and it turns out that Thunderbird can now access your calendar(s) directly without the need for an add-on.

Using the drop-down menus in Thunderbird:
File > New > Calendar
Select 'On the Network' and click on the 'Next' button
Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button
Then enter your logon credentials as normal....and voila....you have full access to your calendar

Thank you pajew, for the heads-up!!!

**Comment by ytdlder** [comment]
can confirm!
TB 91.5.1 (32-bit)
Addon 91.0.4

**Comment by johnk860** [comment]
You are just TOO popular!  Congrats on that.  Hopefully the Google will resolve its issues.

**Comment by lionslair** [comment]
> Okay then, so I've just read in post #373, in a reply from kewisch (the owner of this repository), that the problem has been identified. That an incorrect API was used for the 91.0.3 release, resulting in users having to login again and user rate limits therefore becoming exceeded. kewisch stated that this problem will be rectified for all users shortly and to hang in and be patient...
> 
> Another reply from pajew in the same post #373, simply stated that this add-on is no longer necessary. So I checked this out and it turns out that Thunderbird can now access your calendar(s) directly without the need for an add-on.
> 
> Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar
> 
> Thank you pajew, for the heads-up!!!

Hmm. I tried that adding network calendar but after confirming with google it tried to redirect to some none existent localhost address.

**Comment by Gewerd-Strauss** [comment]
> Then enter your logon credentials as normal....and voila....you have full access to your calendar

Do we use the google-accounts credentials? Because I can _not_ get that to work, try as I might. Just won't accept my password.

**Comment by picorealm** [comment]
> Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar

I tried this however it appears that it is not possible to actually edit or delete any events in the calendar.

**Comment by picorealm** [comment]
> > Then enter your logon credentials as normal....and voila....you have full access to your calendar
> 
> Do we use the google-accounts credentials? Because I can _not_ get that to work, try as I might. Just won't accept my password.

Check your gmail for a "security alert" mail.  You have to confirm before it will allow the account to sync.

**Comment by cozumel68** [comment]
> > Then enter your logon credentials as normal....and voila....you have full access to your calendar
> 
> Do we use the google-accounts credentials? Because I can _not_ get that to work, try as I might. Just won't accept my password.

I'm assuming you entered your full email address (including @gmail.com) in the 'User name ' field...
On my pc, I simply entered my password as normal when prompted
Then I clicked on allow access to calendars, contacts and holidays in the next dialog box
Then when requested, I went to my phone to click on the authorise button
Then I had full access to my calendar....It was that simple for me.....

(The access to the calendar, contacts and holidays part may have come after phone authorisation.  Can't remember which came first).

The phone authorisation is only required if you've enabled multi-factor authentication on Google...

**Comment by Gewerd-Strauss** [comment]
![image](https://user-images.githubusercontent.com/80548781/150921984-bce55cfd-666e-4fa1-82b5-44cd7df3d377.png)

That's the point though. If I press "Search calendars" here, I will be greeted with a password-modal. Enter password - and then it gets rejected immediately.
  
![image](https://user-images.githubusercontent.com/80548781/150922833-b9708fc9-b17a-4a2f-bc08-452597420329.png)

No 2FA-alert is ever send out in time. The ones that are sent out by the event are confirmed, but they reach my phone _after_ Thunderbird claims a wrong password. 

The google accounts activity checker updates a two minutes or so later, asking me if that activity was me - which I confirm, not that it can do anything at that point anymore about an already expired, rejected request.

And then you rinse and repeat.

I never reach the step to add my calendars

**Comment by picorealm** [comment]
> No 2FA-alert is ever send out in time. The ones that are sent out by the event are confirmed, but they reach my phone _after_ Thunderbird claims a wrong password.
> And then you rinse and repeat.
> I never reach the step to add my calendars

I think Thunderbird may be saving your password.  Check saved passwords or config editor to see if it's caching it.

**Comment by Gewerd-Strauss** [comment]
>  Check saved passwords 
Assuming you mean Thunderbird's build-in under "Settings" → "Privacy and Security ( or whatever it is in english)" → Section "Passwords" → Button "Stored Passwords", I definetly don't have it stored there.

> config editor 
Where would I do so? I have no idea.

**Comment by cozumel68** [comment]
> > Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar
> 
> I tried this however it appears that it is not possible to actually edit or delete any events in the calendar.

Have you tried closing and then reopening Thunderbird? (I always shut down and reopen software when I make config changes, as it seems to mostly sort out any potential problems).  I exited Thunderbird after adding the calendar.  When I reopened Thunderbird everything worked fine.   I just tested adding, deleting and editing of events and all is ok....

**Comment by cozumel68** [comment]
> > Check saved passwords
> 
> Assuming you mean Thunderbird's build-in under "Settings" → "Privacy and Security ( or whatever it is in english)" → Section "Passwords" → Button "Stored Passwords", I definetly don't have it stored there.
> 
> > config editor
> 
> Where would I do so? I have no idea.

Via the drop-down menu, go to
Tools > Account Settings
Then select 'Thunderbird Preferences' in the bottom left hand side
This will take you to the 'Preferences' Screen, in the 'General' section
If you scroll down to the bottom, you will find the 'Config Editor'
Before you get to the 'Config Editor' though, there is also an option to clear your cache...(this may or may not be helpful)
In addition, within the 'Preferences' screen you could select 'Privacy & Security' in the left hand pane
On the next screen you will be able to see whether you have any saved passwords stored....

**Comment by Gewerd-Strauss** [comment]
>  In addition, within the 'Preferences' screen you could select 'Privacy & Security' in the left hand pane
> On the next screen you will be able to see whether you have any saved passwords stored....

**Comment by lionslair** [comment]
> > > Check saved passwords
> > 
> > 
> > Assuming you mean Thunderbird's build-in under "Settings" → "Privacy and Security ( or whatever it is in english)" → Section "Passwords" → Button "Stored Passwords", I definetly don't have it stored there.
> > > config editor
> > 
> > 
> > Where would I do so? I have no idea.
> 
> Via the drop-down menu, go to Tools > Account Settings Then select 'Thunderbird Preferences' in the bottom left hand side This will take you to the 'Preferences' Screen, in the 'General' section If you scroll down to the bottom, you will find the 'Config Editor' Before you get to the 'Config Editor' though, there is also an option to clear your cache...(this may or may not be helpful) In addition, within the 'Preferences' screen you could select 'Privacy & Security' in the left hand pane On the next screen you will be able to see whether you have any saved passwords stored....

ok clearing cache seemed to solve it for me

**Comment by leag1234** [comment]
> Another reply from pajew in the same post #373, simply stated that this add-on is no longer necessary. So I checked this out and it turns out that Thunderbird can now access your calendar(s) directly without the need for an add-on.
> 
> Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar

I confirm that this plugin is no longer necessary, as Google Calendar access is now supported natively by Thunderbird.

Tested on Thunderbird 91.5.0

**Comment by ytdlder** [comment]
> > Another reply from pajew in the same post #373, simply stated that this add-on is no longer necessary. So I checked this out and it turns out that Thunderbird can now access your calendar(s) directly without the need for an add-on.

True, BUT: for the Tasks you still need this addon!
(at least I couldn't get it to work with TB integrated Google access)

**Comment by picorealm** [comment]
> > Check saved passwords
> 
> Assuming you mean Thunderbird's build-in under "Settings" → "Privacy and Security ( or whatever it is in english)" → Section "Passwords" → Button "Stored Passwords", I definetly don't have it stored there.
> 
> > config editor
> 
> Where would I do so? I have no idea.

Do you also have a mail set up for the same account?  Maybe it's using the password that's stored for mail?

**Comment by picorealm** [comment]
So I just waited a while and now it's possible to edit events.  So maybe they weren't editable after the initial setup due to syncing.

**Comment by jinnko** [comment]
I couldn't get TB to connect without the app after many attempts and restarts.  So I tried it with a new profile (by launching with `thunderbird -ProfileManager`) and the set up process prompted me to sync contacts and calendars and it's all working now.

**Comment by Gewerd-Strauss** [comment]
Hmm. So clearing the cache only gave me access to an older snapshot of the calendar state - which makes no sense, as it _would_ get the current instance if it were actually querying. So I guess it's storing data regarding events elsewhere.  Events added this morning to the google cal online are not synced. 

![image](https://user-images.githubusercontent.com/80548781/150964793-6827e494-48c1-4182-a1d8-43d97600eb3e.png)


Activating calendars opens the same `Please log on using your google acc`→ `enter pw` → `2fa` → `rate-limit-exceeded`-chain that is the problem right now.


During this I went ahead with a full reinstall of first the add-on, then TB, neither helping as I always end up at the same error.

I suppose the only thing I can do right now is to way for this to be resolved officially, as was outlined in [cozumel's comment](https://github.com/kewisch/gdata-provider/issues/381#issuecomment-1020832425) from a few hours ago, and just move over to use google cal as my primary calendar.


---


@picorealm [regarding this](https://github.com/kewisch/gdata-provider/issues/381#issuecomment-1020948661): Well, It's complicated™.
No. It's definetly not the mail-accounts pw. That one's different to the one on the google account - and getting past the login is not the problem.

**Comment by cozumel68** [comment]
> > > Another reply from pajew in the same post #373, simply stated that this add-on is no longer necessary. So I checked this out and it turns out that Thunderbird can now access your calendar(s) directly without the need for an add-on.
> 
> True, BUT: for the Tasks you still need this addon! (at least I couldn't get it to work with TB integrated Google access)

Oh...I hadn't noticed that Tasks didn't work....okay then....so anyone (including me) who needs that functionality in Thunderbird, need to be patient and wait for the issue to be resolved then, as per the guidance from kewisch in post #373

In the meantime though, I will be using the new(ish) Thunderbird ability to grant full access to gmail calendars (ie to create, edit and delete events), as I outlined a few hours ago.

**Comment by cozumel68** [comment]
> I couldn't get TB to connect without the addon after many attempts and restarts. So I tried it with a new profile (by launching with `thunderbird -ProfileManager`) and the set up process prompted me to sync contacts and calendars and it's all working now.

Ooh...Thanks for that info....I'll try that in a little bit....when I have the time....and hopefully it'll work for me too!!!

**Comment by PDF100** [comment]
(Error 403: rate_limit_exceeded)  Nothing is working.  In TB, adding a new calender, following the single steps, at the end it does not accept the credentials. The 91.0.4 add-on doesn't work neither. I am cut out from all my calender notes. Why they don't change the exceed limit to endless until the bug is fixed?

**Comment by luclaurens** [comment]
just to inform that I face the issue too. 
hoping that the owner of this extension can quicky get an increase of the threshold value from Google

**Comment by PDF100** [comment]
In the mean time our IT expert has managed to add the calenders directly in TB - but this allows only a passive mode to view, not to add or modify.

**Comment by cozumel68** [comment]
> In the mean time our IT expert has managed to add the calenders directly in TB - but this allows only a passive mode to view, not to add or modify.

Did you see my earlier post (4th entry in this thread)?  It gives instructions on how to enable the calendar within Thunderbird so that you can add, delete and modify events.  You still won't be able to use anything in the 'Task' pane though....but events in the calendar should work perfectly...they do for me...

**Comment by MLopez-Ibanez** [comment]
> Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar


I followed these instructions and I have recovered all my calendars except the Tasks. But I can live without that for the moment. In fact, disabling the add-on and using the new native Thunderbird support works better when accepting meeting invitations than the add-on ever did, so I will keep the addon disabled for now.

**Comment by mahWae9i** [comment]
> Using the drop-down menus in Thunderbird: File > New > Calendar Select 'On the Network' and click on the 'Next' button Enter your full Gmail address in the 'User name' field and click on the 'Find Calendars' button Then enter your logon credentials as normal....and voila....you have full access to your calendar

@cozumel68's method works and allows for native editing etc... Thanks!

**Comment by PDF100** [comment]
Thank you for your help!
Now I can use the Calendar functions like before.

*Paul Di Franco*
Export Area Manager
***@***.*** ***@***.***>
_ mob: +39 335 5976177_
_skype: paulghines

GHINES <http://www.ghines.com> 	
facebook <https://www.facebook.com/pages/Ghines-srl/136879926377720> 
youtube <https://www.youtube.com/user/ghinesvideo>

Ghines Group srl- Via Marecchiese 364 - 47923  Rimini (Italy) | tel: 
+390541751080 | www.ghines.com <http://www.ghines.com>|

Informativa per la tutela della privacy.
Le informazioni contenute nella presente trasmissione sono riservate e 
destinate unicamente al nominativo sopra indicato. Qualora riceviate 
erroneamente questa comunicazione vogliate comunicarcelo immediatamente 
e distruggere questo documento. Ogni eventuale abuso della presente 
trasmissione sarà perseguito a norma di Legge.

Information regarding the protection of privacy.
The information contained in this transmission is confidential and 
destined exclusively to the recipient named above. If you received this 
communication by mistake, please inform us immediately and destroy this 
document. Any eventual abuse of this transmission will be prosecuted by law.


Il 25/01/2022 16:06, mahWae9i ha scritto:

**Comment by momist** [comment]
This solution worked fine for me.  Yes, you add a new calendar to your Thunderbird calendar list, enter your full gmail address and Google will then ask for your regular Google password.  Enter that and, by magic, all your calendar entries appear.  You can then delete all the failed calendar listings in the calendar list, and you are good to go.  (at least, I was!  ;-)  )

**Comment by luclaurens** [comment]
Worked for me as well. Thanks

**Comment by kewisch** [comment]
Hey folks, glad some folks found a solution. I'm going to close this in favor of #373. The built in solution uses the CalDAV protocol, there are a few features not available, tasks was noted as one.

**Comment by MLopez-Ibanez** [comment]
> Hey folks, glad some folks found a solution. I'm going to close this in favor of #373. The built in solution uses the CalDAV protocol, there are a few features not available, tasks was noted as one.

Since the other report is locked, I will update whoever is following here. I noticed many synchronization errors with the CalDAV solution that comes with thunderbird. In particular: https://bugzilla.mozilla.org/show_bug.cgi?id=1752109. In some cases, thunderbird would not save an event to the server until I restarted thunderbird and synchronized manually. This is obviously quite bad so I went back to using the add-on except for read-only calendars (no more Error 403 today). I hope this is useful for others.

TL;DR: My conclusion: Use the native Thunderbird solution for read-only Google Calendars and use the add-on for read/write calendars and Tasks.

**Comment by picorealm** [comment]
> TL;DR: My conclusion: Use the native Thunderbird solution for read-only Google Calendars and use the add-on for read/write calendars and Tasks.

Thanks for the info.  Is there any ETA for when this addon will be updated?

**Comment by MrBigGuy** [comment]
I've switched to "TbSync", along with "Provider
      for CalDAV & CardDAV" add-ons.  Requires one line to be added
      to the Config Editor.  Very stable and it works great.  Accepted
      Google login just fine, and even fixed an issue I had with
      Provider for Google Calendar not recognizing local time on
      invites. 
    
    On 1/26/2022 3:55 PM, Pico wrote:
    
    
      
      
        TL;DR: My conclusion: Use the native Thunderbird
          solution for read-only Google Calendars and use the add-on for
          read/write calendars and Tasks.
      
      Thanks for the info. Is there any ETA for when this
        addon will be updated?
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        Triage notifications on the go with GitHub Mobile for iOS or Android.
        
        You are receiving this because you are subscribed to this
        thread.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/kewisch/gdata-provider/issues/381#issuecomment-1022638536",
"url": "https://github.com/kewisch/gdata-provider/issues/381#issuecomment-1022638536",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
